src/key/helper.js

```diff
@@ -106,7 +118,7 @@ export async function getPreferredHashAlgo(key, keyPacket, date = new Date(), us
   let pref_algo = hash_algo;
   if (key) {
     const primaryUser = await key.getPrimaryUser(date, userId);
-    if (primaryUser && primaryUser.selfCertification.preferredHashAlgorithms) {
+    if (primaryUser.selfCertification.preferredHashAlgorithms) {
```

Comment:
May be we could get rid of "if inside if" here. This is up to you.

Reply:
This `if` depends on `primaryUser`, which depends on the other `if` - so we can't merge them, no?

Reply:
I think javascript has lazy if evaluation, so we could create composite condition inside if.

Reply:
Yes but you'd need to create an inline assignment inside the if condition, I don't think that's really cleaner.